design-proposals/passt-feature-gate-iface-api-deprecation.md

```diff
@@ -0,0 +1,176 @@
+spec:
+  configuration:
+    developerConfiguration:
+      featureGates:
+      - Passt
+  ...
+```
+
+2. Specify Passt interface in the VM spec template:
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  name: myvm
+spec:
+  template:
+    spec:
+      domain:
+        devices:
+          interfaces:
+          - name: blue
+            passt: {}
+      networks:
+      - name: blue
+        pod: {}
+  ...
+```
+
+### VM with Passt interface using network binding API:
+
+1. Passt network binding plugin network-attachment-definition should be created, see the following example:
+```yaml
+apiVersion: "k8s.cni.cncf.io/v1"
+kind: NetworkAttachmentDefinition
+metadata:
+  name: kubevirt-passt-binding
+spec:
+  config: '{
+            "cniVersion": "0.3.1",
+            "type": "kubevirt-passt-binding"
+  }'
+```
+
+2. Register Passt network binding network-attachment-definition and sidecar image, by specifying them in Kubevirt configuration, and enable network binding plugin feature:
+```yaml
+apiVersion: kubevirt.io/v1
+kind: KubeVirt
+metadata:
+  name: kubevirt
+  namespace: kubevirt
+spec:
+  configuration:
+    developerConfiguration:
+      featureGates:
+      - NetworkBindingPlugins
+    network:
+      Binding:
+        passt:
+          sidecarImage: quay.io/kubevirt/network-passt-binding:v1.0.0
+          networkAttachmentDefinition: kubevirt-passt-binding
+  ...
+```
+
+3. Specify the registered Passt network binding plugin in VM spec template:
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  name: myvm
+spec:
+  template:
+    spec:
+      domain:
+        devices:
+          interfaces:
+          - name: blue
+            binding:
+              name: passt
+      networks:
+      - name: blue
+        pod: {}
+  ...
+```
+
+
+## Scalability
+
+## Update/Rollback Compatibility
+In Kubevirt v1.3.0: 
+- Existing running VMs that use Passt interface API will keep running with no disruption.
+- Stopped VMs or new ones, will not manage to start and require using the network binding plugin API.
+> **Note:** Passt doesnt support migration.
+
+## Functional Testing Approach
+Kubevirt's Passt network e2e tests shall spin up VMs with Passt interface using network binding plugin API.
+
+## Implementation Phases
+1. Kubevirt v1.2.0: 
+   1. Raise warnings when Passt feature is enabled and Passt interface API is used.
+   2. Declare relevant filed deprecated.
+
+2. Kubevirt v1.3.0:
+   1. Set admission webhook to block VM creation when Passt interface is being used.
+   2. Remove Passt binding implementation from Kubevirt core
+   3. Deprecate Passt feature gate, i.e.: always enabled, no-op.
+
+# Open questions
+## 1. Can we set the webhook to raise warnings at Kubevirt v1.1.1?
+   Kubevirt core Passt binding implementation and Passt network binding plugin already live along each other.
+   We need to do some finlization and set the webhook to raise warnings when Passt feature gate is enabled
```

Comment:
Why the warning needs to be done from the webhook?

Anyway, I would expect to see here a reason why it is better to do it on v1.1.1.

Reply:
> Why the warning needs to be done from the webhook?

This is implementation detail I will remove it, but in general it allows providing better feedback to the user as it will be raised when ` kubectl create/apply` is used.

> Anyway, I would expect to see here a reason why it is better to do it on v1.1.1.

Done

Reply:
> This is implementation detail I will remove it, but in general it allows providing better feedback to the user as it will be raised when ` kubectl create/apply` is used.

I disagree.

Reply:
BTW when enabling deprecated FG you get warning from the webhook following `kubectl apply/patch ..`

Reply:
@EdDev Dissagree with what? with raise warning at webhook level?